Breeding-Insight/bi-api
# [BI-1632] - Unknown Pedigree not displaying for BrAPI-server programs

# Description
**Story:** [BI-1632 -  Unknown Pedigree not displaying for BrAPI-server programs](https://breedinginsight.atlassian.net/browse/BI-1632)

Only valid GIDs can be stored in the backend, so in the case of Unknown or Unknown/Unknown pedigrees, what is stored in the backend for pedigree is null, with the Unknown information being stored in additionalInfo and retrieved upon display. However, BrAPIGermplasmDAO:processGermplasmForDisplay only sets pedigree values, including GERMPLASM_PEDIGREE_BY_NAME, when the pedigree stored is not null. This is not a problem in Breedbase because what is returned is always “NA/NA” so the logic to set pedigree string always triggers. For BrAPI, null was being returned in the case of pedigrees with only Unknown germplasm, so the display pedigree values were not being set.

This change removes the clause checking for non-null pedigree (as it effectively was already always true in the Breedbase case) to enable setting pedigree display values in Unknown and Unknown/Unknown cases.

# Dependencies
bi-web/develop

# Testing
For BB and BrAPI programs
- Import germplasm with unknown parents (both in the form of Unknown, Unknown/Unknown, and <known GID>/Unknown)
- Go to All Germplasm table
- Ensure pedigrees displaying as expected in Pedigree Column
- Go to germplasm details, ensure pedigree also displaying properly there


# Checklist:

- [X] I have run TAF: [_\<please include a link to TAF run>_](https://github.com/Breeding-Insight/taf/actions/runs/3206951585)
